cmake/templates/HPXMacros.cmake.in

```diff
@@ -89,13 +89,15 @@ function(hpx_setup_allocator)
   if(HPX_DISABLE_CUSTOM_ALLOCATOR)
     return()
   endif()
-  if(DEFINED HPX_WITH_MALLOC)
+  if(DEFINED HPX_WITH_MALLOC AND NOT ${HPX_WITH_MALLOC}x STREQUAL x)
```

Comment:
I think this can just be `if(HPX_WITH_MALLOC)`. An empty string should evaluate to false.

Reply:
This actually this happened to me. Not sure what was going on.

Reply:
I think it's just the `DEFINED` that's problematic. `DEFINE HPX_WITH_MALLOC` is true with an empty string, whereas `HPX_WITH_MALLOC` on its own is false with an empty string. I may be missing the point here though...

Reply:
@msimberg: yes, let's ask @aurianer what she had in mind.

Reply:
This is old code. Do you have a test case where this fails now without the extra comparison? If not I think you could just go ahead and do `if(HPX_WITH_MALLOC)`

Reply:
Yes, this does the trick. Thanks!